src/gt4py/next/common.py

```diff
@@ -1035,6 +1035,16 @@ def offset_provider_to_type(
     }
 
 
+def offset_provider_hash(offset_provider: OffsetProvider) -> int:
```

Comment:
Just a suggestion, but in my opinion a more specific name could help here, since this is more like a workaround than the proper way of hashing offset providers 
```suggestion
def key_value_id_hash(mapping: Mapping[Hashable, Any]) -> int:
```

Reply:
Ok, makes sense. What about using some suffix like `hash_offset_provider_unsafe` or `hash_offset_provider_by_value_id` ... ?